src/espidf/ulp_fsm.rs

```diff
@@ -241,8 +240,8 @@ impl Builder {
     }
 
     fn unescape(arg: &str) -> Option<String> {
-        if arg.starts_with("\"-I") && arg.ends_with('\"') {
-            Some(arg[3..arg.len() - 1].replace("\\\"", "\""))
+        if (arg.starts_with("\"-isystem") || arg.starts_with("\"-I")) && arg.ends_with('\"') {
```

Comment:
The original semantics of `unescape` was to:
* Detect if `arg` is an include statement (i.e. starts with either `-I` or `"-I`)
* Strip from it **both** the double quotes **and** the `-I` prefix (which is re-introduced after that)

The new code is wrong: 
* It will strip **only the quotes** from strings starting with `"-I` and `"-isystem`
* It will strip the `-I` from unquoted include args
* It will return `None` for unquoted `-isystem` args

So, if you have
```rust
["-I/foo", "-isystem/bar", "\"-I/quoted_foo\"", "\"-isystem/quoted_bar\"]
```

Your function will return:
```rust
[Some("/foo"), None,  Some("-I/quoted_foo"), Some("-isystem/quoted_bar")]
```

You should either strip `-I` and `-isystem` from both quoted and unquoted strings and unquote the strings. Or NOT strip from both quoted and unquoted strings. And on both cases should return `None` if the string is not an include arg.